sreality/request.go

```diff
@@ -0,0 +1,189 @@
+package sreality
+
+import(
+    "strings"
+    "strconv"
+    "net/http"
+    "io/ioutil"
+    "encoding/json"
+
+    "github.com/grubastik/flat-search/config"
+    "github.com/grubastik/flat-search/email"
+)
+
+const configName = "sreality"
+
+type rangeDefinition struct {
+    min int
+    max int
+}
+
+type UrlParams struct {
+    url string
+    category_main_cb int
+    category_sub_cb *[]int
+    category_type_cb int
+    czk_price_summary_order2 *rangeDefinition
+    estate_age int
+    locality_district_id *[]int
+    locality_region_id *[]int
+    locality_country_id int
+    per_page int
+    usable_area *rangeDefinition
+}
+
+var conf *config.Config
+
+func NewSreality(config *config.Config) *UrlParams {
+    conf = config
+    var urlParameters *UrlParams = new(UrlParams);
+    moduleConfig := config.GetSreality();
+
+    if (moduleConfig.RealityType > 0) {
+        urlParameters.category_main_cb = int(moduleConfig.RealityType)
+    }
+    if (moduleConfig.OperationType > 0) {
+        urlParameters.category_type_cb = int(moduleConfig.OperationType)
+    }
+    if (moduleConfig.RealityOptions != nil) {
+        urlParameters.category_sub_cb = &(moduleConfig.RealityOptions)
+    }
+    if (moduleConfig.Country > 0) {
+        urlParameters.locality_country_id = moduleConfig.Country
+    }
+    if (moduleConfig.Region != nil) {
+        urlParameters.locality_region_id = &(moduleConfig.Region)
+    }
+    if (moduleConfig.District != nil) {
+        urlParameters.locality_district_id = &(moduleConfig.District)
+    }
+    if (moduleConfig.PageResults > 0) {
+        urlParameters.per_page = moduleConfig.PageResults
+    }
+    if (moduleConfig.EstateAge > 0) {
+        urlParameters.estate_age = moduleConfig.EstateAge
+    }
+    if (moduleConfig.Square != nil) {
+        urlParameters.SetUsableArea(moduleConfig.Square.Min, moduleConfig.Square.Max)
+    }
+    if (moduleConfig.Price != nil) {
+        urlParameters.SetPriceRange(moduleConfig.Price.Min, moduleConfig.Price.Max)
+    }
+    if (moduleConfig.Url != "") {
+        urlParameters.url = moduleConfig.Url
+    }
+
+    return urlParameters
+}
+
+func (up *UrlParams) SetUsableArea(min, max int) {
+    up.usable_area = new(rangeDefinition);
+    up.usable_area.min = min
+    up.usable_area.max = max
+}
+
+func (up *UrlParams) SetPriceRange(min, max int) {
+    up.czk_price_summary_order2 = new(rangeDefinition);
+    up.czk_price_summary_order2.min = min
+    up.czk_price_summary_order2.max = max
+}
+
+func (up *UrlParams) GetRequest() (request string) {
+    requestParts := []string{}
+    if up.category_main_cb != 0 {
+        requestParts = append(requestParts, "category_main_cb=" + strconv.Itoa(up.category_main_cb))
+    }
+    if up.category_type_cb != 0 {
+        requestParts = append(requestParts, "category_type_cb=" + strconv.Itoa(up.category_type_cb))
+    }
+
+    if len(*up.category_sub_cb) > 0 {
```

Comment:
If field can be a `nil`, it's good practice to check if `!= nil` first. 

```go
if  up.category_sub_cb !=nil && len(*up.category_sub_cb) > 0 {
```

You would get an error `invalid memory address or nil pointer dereference` if field is nil and you are trying to pointer value.